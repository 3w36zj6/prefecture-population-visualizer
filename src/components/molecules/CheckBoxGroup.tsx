import { CheckBox, CheckBoxProps } from "../atoms/CheckBox";

export interface CheckBoxGroupProps {
  items: CheckBoxProps[];
}

export const CheckBoxGroup: React.FC<CheckBoxGroupProps> = ({ items }) => {
  return (
    <div style={{ display: "flex", flexWrap: "wrap", gap: "10px" }}>
      {items.map((item, index) => (
        <CheckBox
          key={index}
          label={item.label}
          checked={item.checked}
          onChange={item.onChange}
        />
      ))}
    </div>
  );
};
