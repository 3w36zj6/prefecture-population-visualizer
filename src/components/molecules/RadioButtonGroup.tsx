import { RadioButton, RadioButtonProps } from "../atoms/RadioButton";

export interface RadioButtonGroupProps {
  items: RadioButtonProps[];
}

export const RadioButtonGroup: React.FC<RadioButtonGroupProps> = ({
  items,
}) => {
  return (
    <div style={{ display: "flex", flexWrap: "wrap", gap: "10px" }}>
      {items.map((item, index) => (
        <RadioButton
          key={index}
          label={item.label}
          name={item.name}
          value={item.value}
          checked={item.checked}
          onChange={item.onChange}
        />
      ))}
    </div>
  );
};
