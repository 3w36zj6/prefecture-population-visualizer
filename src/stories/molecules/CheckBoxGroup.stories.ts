import type { Meta, StoryObj } from "@storybook/react";
import { fn } from "@storybook/test";
import { CheckBoxGroup } from "./../../components/molecules/CheckBoxGroup";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title: "molecules/CheckBoxGroup",
  component: CheckBoxGroup,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: "centered",
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ["autodocs"],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes
  argTypes: {},
  // Use `fn` to spy on the onClick arg, which will appear in the actions panel once invoked: https://storybook.js.org/docs/essentials/actions#action-args
  args: {},
} satisfies Meta<typeof CheckBoxGroup>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
export const Primary: Story = {
  args: {
    items: [
      {
        label: "CheckBox",
        checked: false,
        onChange: fn(),
      },
      {
        label: "CheckBox",
        checked: true,
        onChange: fn(),
      },
    ],
  },
};
