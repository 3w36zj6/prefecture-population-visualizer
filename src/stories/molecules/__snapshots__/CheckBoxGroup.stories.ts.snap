// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/CheckBoxGroup Primary smoke-test 1`] = `
<div style="display: flex; flex-wrap: wrap; gap: 10px;">
  <label>
    <input type="checkbox">
    CheckBox
  </label>
  <label>
    <input type="checkbox"
           checked
    >
    CheckBox
  </label>
</div>
`;
